cpp/include/cudf_test/column_wrapper.hpp

```diff
@@ -126,80 +131,59 @@ struct fixed_width_type_converter {
- * @tparam ElementFrom The element type used to create elements of type `ElementTo`
- * @tparam InputIterator Iterator type for `begin` and `end`
- * @param begin Beginning of the sequence of elements
- * @param end End of the sequence of elements
- * @return rmm::device_buffer Buffer containing all elements in the range `[begin,end)`
- */
-template <typename ElementTo,
-          typename ElementFrom,
-          typename InputIterator,
-          typename std::enable_if_t<not cudf::is_fixed_point<ElementTo>()>* = nullptr>
-rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
-{
-  static_assert(cudf::is_fixed_width<ElementTo>(), "Unexpected non-fixed width type.");
-  auto transformer     = fixed_width_type_converter<ElementFrom, ElementTo>{};
-  auto transform_begin = thrust::make_transform_iterator(begin, transformer);
-  auto const size      = cudf::distance(begin, end);
-  auto const elements  = thrust::host_vector<ElementTo>(transform_begin, transform_begin + size);
-  return rmm::device_buffer{elements.data(), size * sizeof(ElementTo), rmm::cuda_stream_default};
-}
-
-/**
- * @brief Creates a `device_buffer` containing the elements in the range `[begin,end)`.
- *
- * @tparam ElementTo The element type that is being created (`fixed_point` specialization)
- * @tparam ElementFrom The element type used to create elements of type `ElementTo`
- * (non-`fixed-point`)
- * @tparam InputIterator Iterator type for `begin` and `end`
- * @param begin Beginning of the sequence of elements
- * @param end End of the sequence of elements
- * @return rmm::device_buffer Buffer containing all elements in the range `[begin,end)`
+ * @brief Copy the iterator range to device memory.
+ * If the iterators are tagged device iterators, the copy is performed directly.
+ * Else, the copy is performed on the host and then copied to device.
  */
-template <typename ElementTo,
-          typename ElementFrom,
-          typename InputIterator,
-          typename std::enable_if_t<not cudf::is_fixed_point<ElementFrom>() and
-                                    cudf::is_fixed_point<ElementTo>()>* = nullptr>
-rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
+template <typename RepType, typename InputIterator>
+inline rmm::device_buffer make_device_elements(InputIterator begin, InputIterator end)
 {
-  using RepType        = typename ElementTo::rep;
-  auto transformer     = fixed_width_type_converter<ElementFrom, RepType>{};
-  auto transform_begin = thrust::make_transform_iterator(begin, transformer);
-  auto const size      = cudf::distance(begin, end);
-  auto const elements  = thrust::host_vector<RepType>(transform_begin, transform_begin + size);
+  auto const size = cudf::distance(begin, end);
+#ifdef __CUDACC__
+  if constexpr (std::is_same<typename thrust::iterator_system<InputIterator>::type,
+                             thrust::device_system_tag>::value) {
```

Comment:
Suggesting `std::is_same_v<T, U>` (C++17) instead of `std::is_same<T, U>::value`.
```suggestion
  if constexpr (std::is_same_v<typename thrust::iterator_system<InputIterator>::type,
                               thrust::device_system_tag>) {
```